object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -213,11 +219,32 @@ private boolean isOwningAssignmentLhs(Element elem) {
    * Checks for cases where we do not need to ensure that a method invocation gets pseudo-assigned
    * to a variable / field that takes ownership. We can skip the check when the invoked method's
    * return type is {@link org.checkerframework.common.returnsreceiver.qual.This}, the invocation is
-   * a super constructor call, or the method's return type is annotated {@link NotOwning}
+   * a super constructor call, the method's return type is annotated {@link NotOwning}, or the
+   * receiver parameter is in the {@code defs} and has @MustCallChoice annotation.
    */
-  private boolean shouldSkipInvokePseudoAssignCheck(Tree callTree) {
+  private boolean shouldSkipInvokePseudoAssignCheck(
+      Node node, Set<ImmutableSet<LocalVarWithTree>> defs) {
+    Tree callTree = node.getTree();
+    List<String> mustCallVal = typeFactory.getMustCallValue(callTree);
+    if (mustCallVal.isEmpty()) {
+      return true;
+    }
+    // In this case, we are handling method invocation nodes that are not assigned to a local
+    // variable node but the receiver of the method is in the defs
+    if (callTree.getKind() == Tree.Kind.METHOD_INVOCATION
+        || callTree.getKind() == Tree.Kind.NEW_CLASS) {
+      LocalVariableNode mustCallChoiceParam = getLocalPassedAsMustCallChoiceParam(node);
+      if (mustCallChoiceParam != null) {
+        if (isVarInDefs(defs, mustCallChoiceParam)) {
```

Comment:
These lines could be simplified to `return isVarInDefs(defs, mustCallChoiceParam)`